mitodl/bootcamp-ecommerce
# Facebook pixel tracking

#### What are the relevant tickets?
Closes #182

#### What's this PR do?
- Adds Facebook pixel tracking using the `react-facebook-pixel` plugin
- Sends pageview info to FB on the payment page
- Sends payment info to FB on successful order payment

#### How should this be manually tested?
- Create your own Facebook pixel id at https://www.facebook.com/events_manager/pixel/
- Set `FACEBOOK_PIXEL_ID=<your_id>` in your `.env` file
- Go to the `/pay` page with the network tab open.  There should be a request made to Facebook with parameters:
    - `ev=PageView`
    - `dl=<your local bootcamp payment url>`
    - `id=<your_pixel_id>`
- Start a payment for your klass.  
- In the admin console, set the order status to `Fulfilled`
- Go to `../pay?status=receipt&order=<order_id>` with the network tab open.  There should be a request made to Facebook with parameters:
    - `ev=Purchase`
    - `cd[value]=<amount_paid>`
    - `cd[currency]='USD'`
- Eventually, pageview and purchase events should show up at https://www.facebook.com/events_manager/pixel/
